filament/backend/src/vulkan/VulkanPipelineCache.h

```diff
@@ -77,13 +77,15 @@ class VulkanPipelineCache : public CommandBufferObserver {
     // The RasterState POD contains standard graphics-related state like blending, culling, etc.
     // Note that several fields are unused (sType etc) so we could shrink this by avoiding the Vk
     // structures. However it's super convenient just to use standard Vulkan structs here.
-    struct alignas(8) RasterState {
+    #pragma pack(push, 1)
+    struct UTILS_PACKED RasterState {
```

Comment:
this doesn't have performance impact? -- if fields are not aligned to their natural alignement it can cause all sorts of issues -- it's sometimes okay, but it's hard to tell just looking at this structure.

I think it's better to show the padding explicitly and have a static_assert<> on the size.

Reply:
I'm using structures from the Vulkan header inside this struct, so I can't make padding explicit or rearrange their members.  I do not store these in a contiguous array, so I don't think this will impact performance in a significant way.

Reply:
and you never pass a pointer to them to vk (or to a piece of code that wouldn't expect the packed format)?

Reply:
Good point, I decided that using the Vk spec structs for state tracking is just plain wrong, added a commit to fix this.